service/controller/chart/resource/release/desired.go

```diff
@@ -49,11 +49,9 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 			return nil, microerror.Mask(err)
 		}
 
-		valuesMD5Checksum = fmt.Sprintf("%x", md5.Sum(valuesYAML))
-	} else {
-		// We need to pass empty values in ValueOverrides to make the install
-		// process use the default values and prevent errors on nested values.
-		valuesYAML = []byte("{}")
+		// MD5 is only used for comparison but we need to turn off gosec or
+		// linting errors will occur.
+		valuesMD5Checksum = fmt.Sprintf("%x", md5.Sum(valuesYAML)) // #nosec
```

Comment:
IMO we should check how we can make it compliant with the linter. But now I am wondering what `// #nosec` is doing. Is that the same as `// nolint:nosec`?

Reply:
@tomahawk28 I'll let you reply to Tim on the other issues.

For this linting issue I checked with Zach and he is OK with us using MD5 since its just for change detection. 

Jihyuk can you check if `nolint:nosec` works? As I think that is clearer.

https://github.com/giantswarm/chart-operator/pull/437#discussion_r413910860

Reply:
I will give it a try, with an upper suggestion from Tim as well.

Reply:
> service/controller/chart/resource/release/desired.go:54:41: G401: Use of weak cryptographic primitive (gosec)

Okay, it thinks that md5 is weak enough to use it for our annotations, but we merely use it to check changes.